### PR TITLE
[DOC] Improve topic replication factor change procedure

### PR DESCRIPTION
**Pull request by fvaleri**
This is still based on the original secure cluster setup (mTLS), as it is common in production. The UTO is not able to reconcile a replication factor change done in Kafka, so I also show how to fix the reconciliation error.

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
I think you should stick with how it was and update only the parts relevant to UTO. Not rewrtie the commands _your way_.

**Comment by fvaleri** [comment]
> I think you should stick with how it was and update only the parts relevant to UTO. Not rewrtie the commands _your way_.

Wdym? Most of these commands where missing.

**Comment by scholzj** [comment]
The commands as you wrote them will anyway not work for half of the readers. We also do not have the resources to test the commands every release. So I think you should stick with how it was and update only what is relevant to UTO.

**Comment by fvaleri** [comment]
These commands should work for users in that specific use case. Other can easily adjust. I don't think the current procedure makes it particularly easy to do that operation, as you have to jump through various sections, but if you are happy with it.

File: documentation/modules/configuring/proc-changing-topic-replicas.adoc [hunk 1]
```diff
@@ -130,4 +130,12 @@ bin/kafka-topics.sh --bootstrap-server
...
\ No newline at end of file
+----
+
+. Finally, patch the `KafkaTopic` custom resource to match the new replication factor, and let the operator do the rest
+[source,shell,subs=+quotes]
+----
+kubectl patch kt my-topic --type merge -p '
+  spec:
+    replicas: 3'
+----
```
Comment (scholzj) [inline documentation/modules/configuring/proc-changing-topic-replicas.adoc:141]: I would just write it in a generic way:

```suggestion
. Finally, update the `KafkaTopic` resource to match the new replication factor (in `.spec.replicas`).
```

Many users use various tooling and cannot do a patch command like this.

**Review by scholzj — CHANGES_REQUESTED** [review]

**Review by scholzj — APPROVED** [review]

File: documentation/modules/configuring/proc-changing-topic-replicas.adoc [hunk 2]
```diff
@@ -130,4 +130,18 @@ bin/kafka-topics.sh --bootstrap-server
...
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: my-topic
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 3
+  replicas: 3
+----
```
Comment (PaulRMellor) [inline documentation/modules/configuring/proc-changing-topic-replicas.adoc:147]:
```suggestion
. Finally, edit the `KafkaTopic` custom resource to change `.spec.replicas` to 3, and then wait the reconciliation.
+
[source,shell,subs=+quotes]
----
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaTopic
metadata:
  name: my-topic
  labels:
    strimzi.io/cluster: my-cluster
spec:
  partitions: 3
  replicas: 3
----
```

Comment (PaulRMellor) [reply documentation/modules/configuring/proc-changing-topic-replicas.adoc:147]: It needs that + on line 136 to bring the example config inline with the step.

**Review by PaulRMellor — APPROVED** [review]
Thanks Fede. Looks good

File: documentation/modules/configuring/proc-changing-topic-replicas.adoc (hunk 2 above)
Comment (fvaleri) [reply documentation/modules/configuring/proc-changing-topic-replicas.adoc]: Thanks Paul.

**Comment by scholzj** [comment]
Thanks for the PR.

**Pull request merged**